Make website responsive

this-is-rafa opened the pull request:
Fixes #117 

I stumbled across the project yesterday via HN and I noticed it was very hard to read.

This PR makes the website responsive without being very opinionated on much else.

After I started a proposal was posted for a complete redesign. I think that's a good idea but will likely require editing every single individual post. This is more of a stop-gap solution that looks more elegant while being functional.

Tested in Mac OS Firefox, Safari, Chromium, Android Firefox, Chromium, and iOS Safari, Firefox, Chrome.

![desktop](https://user-images.githubusercontent.com/12503367/111494063-88d9c180-8714-11eb-9357-9b8acf80bd76.png)

![mobile](https://user-images.githubusercontent.com/12503367/111494070-8a0aee80-8714-11eb-918d-067f401cf195.png)

rhatdan commented:
@TomSweeneyRedHat @vrothberg PTAL

TomSweeneyRedHat commented:
@rhatdan ack and apologies for the delay.  I want to sit down when I have a solid dedicated chunk of time to review and I've not found it yet.  Will get this reviewed by Monday at the absolute latest.

robbyoconnor commented:
Great work @this-is-rafa

vrothberg reviewed (commented):
Thanks a lot for the contribution!

I am terribly unqualified to review. I live in the terminal and can barely remember how to turn text bold in html.

robbyoconnor commented:
@vrothberg -- I'll try to review this -- not the greatest at it, but I'm able to rough it

TomSweeneyRedHat commented on assets/css/style.scss:
```diff
@@ -3,98 +3,267 @@
+);
+
+* {
+  box-sizing: border-box;
```
Oh, very nice.

TomSweeneyRedHat commented on index.md:
```diff
@@ -2,19 +2,6 @@
-</script>
-<link rel="shortcut icon" type="image/x-icon" href="/images/favicon.ico">
-</head>
-
```
Just curious why you moved this instead of leaving it here?

robbyoconnor replied:
In my opinion, it's better to have markdown in Markdown and HTML moved into the layout files.

this-is-rafa replied:
This is a separation of concerns thing. The way the template works the .md files should ideally be content only, while all layout and scripts should be placed in the layout files (/_layouts).

Because the Google Analytics script was only in index.md and not in the layout file, it was only being called on the homepage and thus would not report stats anywhere else.

It also caused a duplication of the `<head>` tags, as index.md is loaded into the default layout. While browsers are very robust in parsing just about any HTML thrown at them, it's not proper HTML.

There are more instances of this in the template and posts which I may tackle in other PRs, but this one really stood out as egregious so I fixed it here.

TomSweeneyRedHat replied:
@this-is-rafa thanks for the 411 and education.  That all makes sense.

robbyoconnor commented:
Might beneficial to add a `Gemfile`

```ruby
source 'https://rubygems.org'

gem "github-pages", group: :jekyll_plugins
```

Be sure to also track the `Gemfile.lock` file. 

We can add to the `README.md` to test changes do: `bundle exec jekyll s`

robbyoconnor reviewed (approved):
Okay, I reviewed this and tested it out on both my laptop and mobile and it is very good. 

![image](https://user-images.githubusercontent.com/23088/111854551-23452b00-88f6-11eb-8f14-a575ea9c98ad.png)

Click the Hamburger menu and this shows: 

![image](https://user-images.githubusercontent.com/23088/111854569-33f5a100-88f6-11eb-9505-370fa90e2f17.png)

On the desktop, it's like @this-is-rafa showed, so no need to show that -- this fixes the concern I had -- again thanks @this-is-rafa -- this is the spirit of Open Source in action.

robbyoconnor commented:
Added a Gemfile in #370 so that people can test locally.

TomSweeneyRedHat commented:
@this-is-rafa I like the look an feel quite a bit, but do have some comments.  I'm not quite sure where in the code they apply, so thought I'd just do a comment here.

If I click on one of the links in the left bar, say "Join Community", it brings me to that page just fine.  However, when I click on the "podman" logo in the top left, I don't go back to the main page, rather the "Join Community" page refreshes.  Maybe that's due to using jekyll to serve the site locally?

The words and links from the bar in the current page have been removed.  "Manage pods, containers, and container images." above the buttons, and below the buttons the words:  

"This project is maintained by the containers organization.
Subscribe to the blog feed."

Was this an oversight or a style thing?  If you really think it's bad form to have them in the bar, we'll need to get them near the top of the main page somewhere.

When I go to a subpage like "Join Community", the logo at the top of the page is centered.  Can that be aligned left similar to the main page?

It might be that I've tired eyes, but I found the text on the left side in the purple bar to be a little light and a bit hard to read.  Can that be brightened a bit?

All in all, a nice update, tyvm!

robbyoconnor commented:
@TomSweeneyRedHat he didn't change anything but the layout

this-is-rafa commented:
@TomSweeneyRedHat Let me address these one by one:

> If I click on one of the links in the left bar, say "Join Community", it brings me to that page just fine. However, when I click on the "podman" logo in the top left, I don't go back to the main page, rather the "Join Community" page refreshes. Maybe that's due to using jekyll to serve the site locally?

Yes, exactly. When pushed to GitHub those links resolve correctly. I've added a trailing slash which makes it work locally also.

> The words and links from the bar in the current page have been removed. "Manage pods, containers, and container images." above the buttons, and below the buttons the words:
> 
> "This project is maintained by the containers organization.
> Subscribe to the blog feed."
> 
> Was this an oversight or a style thing? If you really think it's bad form to have them in the bar, we'll need to get them near the top of the main page somewhere.

This was indeed an oversight. The project text was in a conditional statement that would only appear once published to GitHub. I've gone ahead and removed the conditional and styled it accordingly.

> When I go to a subpage like "Join Community", the logo at the top of the page is centered. Can that be aligned left similar to the main page?

I can't seem to recreate this. The logo should be centered everywhere.

> It might be that I've tired eyes, but I found the text on the left side in the purple bar to be a little light and a bit hard to read. Can that be brightened a bit?

I've increased the contrast.

![podman website](https://user-images.githubusercontent.com/12503367/112027261-64fbee80-8b0d-11eb-853c-d084cf5526bf.png)

TomSweeneyRedHat commented:
As far as the top logo is concerned with left vs center alignment, I may have been looking at the current site, not the WIP one when I was looking at the "main page".  So, would it be possible to have the logo at the top of each page left align rather than center align?  I have the tendency to unreasonably widen web browsers and it looks odd to have the logo centered in those instances.

TomSweeneyRedHat commented:
I'll take another look later today, but from a brief scan of your pictures, it looks good!

feloy commented:
@TomSweeneyRedHat could you please have a look at this PR?

TomSweeneyRedHat commented:
Thanks @benoitf for the assist, just returned from PTO this morning.
